sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -447,7 +447,16 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       // Create the table if the table didn't exist.
       if (!tableExists) {
         val schema = JdbcUtils.schemaString(df, url)
-        val sql = s"CREATE TABLE $table ($schema)"
+        // To allow certain options to append when create a new table, which can be
+        // table_options or partition_options.
+        // E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
+        val createtblOptions = {
+          extraOptions.get("jdbc.create.table.options") match {
```

Comment:
Thanks Sean. Actually, here I have a little bit hesitation. For example, "mergeSchema" which may not be so that similar to the other option name (prefixed with "spark"). 
```
val mergedDF = spark.read.option("mergeSchema", "true").parquet("data/test_table")
```

How about to use some short name as "createTableOptions"?